crates/satchel/src/lib.rs

```diff
@@ -1,36 +1,89 @@
+//! Test registration and discovery primitives for custom harnesses built on Satchel.
+//!
+//! This crate collects metadata for functions annotated with [`macro@test`] and
+//! [`macro@bench`], storing them in a linkme-powered distributed slice. Harnesses invoke
+//! [`get_tests!`] to enumerate the [`TestCase`] entries for the current crate, inspect
+//! [`TestCase::case_attributes`] for custom markers, and run or filter tests and benchmarks as
+//! needed. See the project README for end-to-end examples, including CTest integration and
+//! `libtest-mimic` runners.
 #![no_std]
 
-pub use satchel_macro::{bench, test};
+/// Re-export of the attribute macro used to register unit tests with the Satchel harness.
```

Comment:
```suggestion
/// The main macro of the satchel crate used to register unit tests with the Satchel harness.
/// It works like the standard Rust [test macro](https://doc.rust-lang.org/rust-by-example/testing/unit_testing.html#unit-testing), but can be used with custom test harnesses.
///
/// Satchel supports the standard `#[ignore]` and `#[should_panic="..."]` attributes, as well as custom arguments that can be freely interpreted by the test harness via `#[test(...)]`
```

It doesn't matter that this is a re-export, as users are supposed to use the macro from here.
We do need to document what the macro actually supports, as macro behavior is not otherwise discoverable from the API.